### ISSUE TITLE
Update Issue template for jekyll repo

### ISSUE THREAD

**Issue by ericbrasiln** [programminghistorian/jekyll#2820]
# jekyll: Issue Template

  The template was last updated in 2016 (Latest commit [4a8ad0f](https://github.com/programminghistorian/jekyll/commit/4a8ad0f3c4069221c5ea9af18ce36345588a9669) on 27 Jun 2016) and its syntax is outdated.

The current [ISSUE_TEMPLATE.md](https://github.com/programminghistorian/jekyll/blob/gh-pages/.github/ISSUE_TEMPLATE.md) looks like this:

```
If you are reporting a problem with a lesson, please try to provide us with the following information:  

- The full title of the lesson
- The system you are using (Mac, Linux, Windows)
- Version numbers of the relevant software you are using
- The exact steps you took that caused the problem  

If you'd like, you can delete this template text before posting your issue.
```

My update suggestion:

```
---
name: Basic Template
about: Simple issue template
title: Name your issue
---

## If you are reporting a problem with a lesson, please try to provide us with the following information:  

- [ ] The full title of the lesson
- [ ] The system you are using (Mac, Linux, Windows)
- [ ] Version numbers of the relevant software you are using
- [ ] The exact steps you took that caused the problem.

## If you are reporting a technical problem in the Programming Historian site or in the repository, please try to provide us with the following information:

- [ ] Link and page title
- [ ] A brief description of the problem
- [ ] Expected behavior
- [ ] Proposed solution (if any)

*If you'd like, you can delete this template text before posting your issue.*
```

@programminghistorian/technical-team

**Comment by anisa-hawes** [comment]
Dear @ericbrasiln,

Thank you for initiating this update!

I've created [a branch](https://github.com/programminghistorian/jekyll/tree/Issue-2820) to implement your suggestion, including the header. 

I've added in a section which sets out some guidance for setting ourselves objectives that are specific, measurable, attainable, relevant and time-bound (SMART). This relates to a recent conversation between @acrymble, @GeeAlmeida and I about how we use Issues in our daily work on the project. I hope other members of the project team might find this useful as well?

```
## If you are opening this Issue to begin a task or set an objective, please try to use the SMART criteria:

- [ ] **Specific** Which specific area of the project will this work improve? Which need will this meet?
- [ ] **Measurable** How can progress be measured or quantified?
- [ ] **Attainable** What can be achieved using available resources?
- [ ] **Relevant** How does this work contribute towards our broader shared objectives?
- [ ] **Time-Bound** Can you estimate a deadline or define an end-point?
```

I've also added in a sentence that directs contributors to our Code of Conduct:

`**All contributors are reminded that participation in discussion on our community forums is dependant upon respect for one another. Please read our repository [Code of Conduct](https://github.com/programminghistorian/jekyll/blob/gh-pages/CODE_OF_CONDUCT.md).**`

You can see the updated template [here](https://github.com/programminghistorian/jekyll/blob/Issue-2820/.github/ISSUE_TEMPLATE.md). I note that we are still using the old version of issue templates. Does this header bring us up to date with the new issue template workflow? Or is another adjustment needed?

**Comment by anisa-hawes** [comment]
Another thought is that I remember you showing me how an issue template-chooser would work, so that when a new issue is created users can select the template that is appropriate to the Issue they want to open. 

[These instructions](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser) explain how we could set this up by adding a `config.yml` file to the .github/ISSUE_TEMPLATE folder.

What do you think?

**Comment by ZoeLeBlanc** [comment]
We discussed this issue at the @programminghistorian/technical-team meeting today and one thought was if we should have Issue Templates in multiple languages. I realize this might create complications but at the moment we assume anyone can use our English template which seems out of step with our ethos. Wondering if this config.yml approach would allow us to start building more multilingual infrastructure and not just content?

**Comment by ericbrasiln** [comment]
Hi @anisa-hawes and @ZoeLeBlanc 

I think we should have a set of issues templates in all four languages in the `ph-submissions` repository. I had already proposed this to Daniel Alves and Anisa. At the time, we understood that for the `jekyll` repository it would be better to keep only one template in English, since most of the issues here are related to the entire project and generally require the understanding and participation of members of various teams.

I created a repository with the issue template configuration in all languages as an example that we can use in ph-submissions. You can see it [here](https://github.com/ericbrasiln/issue_templates_test/issues) and click on **new issue** to test it.

The file structure is created in [.github/ISSUE_TEMPLATES/](https://github.com/ericbrasiln/issue_templates_test/tree/main/.github/ISSUE_TEMPLATE) and the settings are added to [config.yml](https://github.com/ericbrasiln/issue_templates_test/blob/main/.github/ISSUE_TEMPLATE/config.yml).

I included the possibility of creating a blank issue (without using any template), useful links at the end of the template list, and also defined automatic labeling.

I believe this method can help to create the multilingual infrastructure that Zoe mentioned.

What do you think?

I added @ZoeLeBlanc and @anisa-hawes as collaborators in the repository with the examples.

https://user-images.githubusercontent.com/58128421/212340386-76722490-627b-4f40-b0e5-f9ba6b7afcf5.mp4

**Comment by ericbrasiln** [comment]
For jekyll we could create an issue template for technical problems on the site and repository and another for problems with specific lessons, each of them with specific labels.

> Hi @anisa-hawes and @ZoeLeBlanc
> 
> I think we should have a set of issues templates in all four languages in the `ph-submissions` repository. I had already proposed this to Daniel Alves and Anisa. At the time, we understood that for the `jekyll` repository it would be better to keep only one template in English, since most of the issues here are related to the entire project and generally require the understanding and participation of members of various teams.

**Comment by anisa-hawes** [comment]
Thank you, @ericbrasiln. That makes good sense because within Jekyll, we are using our shared language (English).

Let's collaborate to update the ph-submissions templates at a different moment. I'm working on a slight re-configuration of the labels I mentioned last time we spoke, and also would like to suggest that we don't open Issues for proposals (rather wait until the stage that the lesson files have been submitted).

**Comment by ericbrasiln** [comment]
Hi @anisa-hawes 
Great, I'm happy to help.
I'm going to take a look at your [branch](https://github.com/programminghistorian/jekyll/tree/Issue-2820) and at #2822 later this week.
